src/commands/test.rs

```diff
@@ -40,35 +36,64 @@ pub(crate) enum Command {
             help = "If set to 0 it will run indefinitely, If not set defaults to 1 run."
         )]
         reruns_wanted: Option<u8>,
+        #[arg(
+            long = "withdraw",
+            short = 'w',
+            default_value_t = false,
+            help = "If set, the funds will be withdrawn after each run."
+        )]
+        withdraw: bool,
+        #[arg(
+            long = "sleep",
+            short = 's',
+            default_value_t = 1,
+            help = "Sleep interval between each rerun"
+        )]
+        sleep_secs: u64,
+    },
+    #[clap(
+        about = "Gas Measurements for the zkStack Chain.",
+        visible_alias = "gs"
+    )]
+    GasScenario {
+        #[clap(long = "tps", required = true)]
+        tps: u64,
+        #[clap(
+            long = "amount",
+            short = 'a',
+            required = true,
+            help = "Amount of BaseToken to deposit, 20% more will be deposited.\nThat extra 20% will remain in the main wallet,\nthe rest will be redistributed to random wallets"
+        )]
+        amount: f32,
+        #[arg(
+            long = "rounds",
+            short = 'r',
+            default_value_t = 1,
+            help = "Amount of times to run the program in a loop, it defaults to 1. Max is 255"
+        )]
+        rounds: u8,
     },
 }
 
 impl Command {
     pub async fn run(self, cfg: ZKSyncConfig) -> eyre::Result<()> {
+        let (zk_wallet, l1_provider, l2_provider) = get_wallet_l1_l2_providers(cfg)?;
+        let base_token_address = l2_provider.get_base_token_l1_address().await?;
+        let (base_token_decimals, base_token_symbol) =
+            get_erc20_decimals_symbol(base_token_address, &l1_provider).await?;
+
+        let mut reruns = 0;
+        let mut current_reruns: u32 = 1;
         match self {
             Command::LoadTest {
                 number_of_wallets,
                 amount,
                 reruns_wanted,
+                withdraw,
```

Comment:
I think we should move each subcommand logic to a new file to clean a bit the code, it's getting very long. There are some examples in other commands on how to do this.

Reply:
That will be cleaner, but since we only have ~350 lines in the test.rs file, we can leave it as is for now. We can separate more if we add more commands in the future.